README.md

```diff
@@ -1,166 +1,171 @@
 };
 
-// Use the plugin and pass along the configuration
-Vue.use(VueGAPI, apiConfig);
-```
+// Use the plugin and pass along the configuration  
+Vue.use(VueGAPI, apiConfig);  
+```  
 
-Exposes the `$getGapiClient` on the Vue instance that returns a promise containing the initialised instance of the Google API client.
-See full list of options for apiConfig object (here)[https://developers.google.com/identity/sign-in/web/reference] under gapi.auth2.SignInOptions
+For a complete list of the [apiConfig options](https://developers.google.com/identity/sign-in/web/reference) see the gapi.auth2.SignInOptions section.
 
-```html
+The plugin exposes `$gapi` on the Vue instance as entry point to the plugin functions and variables. The variable `$gapi.getGapiClient()` returns a promise containing the initialised instance of the Google API client.
+  
+```html  
 <script>
-
-export default {
-  name: 'my-component',
-
-  methods: {
+export default {  
+  name: 'my-component',  
+  methods: {  
     login () {
-      this.$getGapiClient()
-        .then(gapi => {
-          // gapi.sheets.spreadsheet.get(...)
-          // ...
-        })
+      this.$gapi.getGapiClient()  
+        .then(gapi => {  
+          // gapi.sheets.spreadsheet.get(...
+        }
+        .catch(err => {
+	      // Handle errors
+        }
     }
-  },
-
+  }
 }
-</script>
-```
-
-## Login
-
-This will shortcut the login process
-
-```html
-<script>
-export default {
-  name: 'login-shortcut',
-
-  methods: {
-    login () {
-      this.$login()
+</script>  
+```  
+  
+## Login  
+Initiate the login process.
+  
+```html  
+<script>  
+export default {  
+  name: 'login',
+  methods: {  
+    login () { 
+       this.$gapi.login()
+         .then(() => {
+           // Any actions needed after login
+         }
+         .catch(err => {
+	       // Handle errors
+         }
     }
   }
-
 }
-
 </script>
-```
-
-## Logout
-
-This will shortcut the logout process
-
-```html
-<script>
-export default {
-  name: 'logout-shortcut',
-
+```  
+  
+## Logout  
+Initiate the logout process.
+  
+```html  
+<script>  
+export default {  
+  name: 'logout',
   methods: {
-    logout () {
+    logout () { 
       this.$logout()
+        .then(() => {
+          // Any actions needed after logout
+        }
+        .catch(err => {
+          // Handle errors
+        }
     }
   }
-
 }
-
-</script>
-```
-
-## isAuthenticated
-
-This will shortcut and check if your user is authenticated will return a boolen
-
-```html
-<script>
+</script>  
+```  
+  
+## isAuthenticated  
+Use isAuthenticated to check if a user is authenticated. It returns true if the user is authenticated, otherwise false.
+
+```html  
+<script>  
 export default {
-  name: 'login-shortcut-check',
-
+  name: 'auhtenticated-check',
   methods: {
     login () {
-      if (this.$isAuthenticated() !== true) {
-        this.$login()
+      if (this.$gapi.isAuthenticated()) {  
+        this.$login() 
```

Comment:
Should the $login method be called from under $gapi.login? if this is the direction we are going.

Reply:
It is part of the vue-gapi plugin. As you pointed out in your previous comment it is not a good idea to pollute the Vue instance with a whole lot of instance methods from the same plugin. It also makes the client code more readable and understandable for someone that is not acquainted with the vue-gapi plugin.